### DISCUSSION TITLE
DisallowConcurrentExecution being ignored?

### DISCUSSION THREAD

**Discussion by moyers** [quartznet/quartznet#2327] [General]
I have a quartz job that basically uses .net Process class to call various commands/exe.  We sometimes use it to call a bash script. Within the bash script is a call to a python script. We do this because we have some other tasks that need to be done before the python script is called. Now the quartz job class has the "DisallowConcurrentExecution" attribute in it and it seems to  work most of the time for most commands/exe. Example, our script runs for 3 hours and we have it schedule to run every hour. When the schedule hits, and the exe is still running, the trigger and/or job events won't fire. However, on the line that calls the py script, we have a flock. When that bash script gets executed, we noticed in the logs that the trigger listener events are being fired. So, it seems that the DisallowConcurrent is not working. If I remove the flock, it seems to work like it should. I guess my question is; does anyone know why that is happening? We may have to leave the flock for other reasons, so if there is a workaround, it would be ideal. It's a long shot since this is so specific, but hoping someone has at least some ideas. Thanks!

**Comment by jafin** [comment]
Apologies I don't know what a `flock` command is.

Do you have any logs for the job, can you see the full timeline of the job, start/end and how that correlates with the shell execution time?

Just curious if you can note that the job is still 'active'.  and the flock command isn't doing something like returning early back to the job and hence completing the job (allowing the next one to run)

We used to have a shell out to a windows batch, and although we expected the cmdshell to wait for the return, it actually triggered asynchronously, returning back to quartz while still running a forked process.  I am just wondering if this may be a similar issue?

Can you wrap some logging around your shell command to verify the job is still 'active' and that the cmd hasn't returned back early to the job?

**Reply by moyers** [reply]
flock (file lock) is a linux command. It puts a lock on a file to prevent other resources from executing/accessing while it is already running/in use. It's not supposed to release until the file that has the lock as either ended its process or an explicit command is sent to release the lock. However, what you have suggested are some good ideas to try. I will need to dig in and see how the actual flock is behaving. Thanks

**Reply by moyers** [reply]
Update for anyone that comes across this:
This was not an issue after all. User error reading the logs and assuming jobs that were already running were trying to start again (concurrent). So, the attribute to prevent concurrent jobs does work even when using it as I described in the original post.
